### PR TITLE
General cog

### PR DESCRIPTION
**Pull request by SeraphimSera**
I suck at programming xd
Pls to fix?

### CODE REVIEW

**Review by SeraphimSera — COMMENTED** [review]
fuck i meant to commit the bot constructor stuff to the to the master branch lmao

File: cogs/General.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+import re
+
+class General(commands.Cog):
+    """Random commands and stuffs go here, kthx"""
+
+    def __init__(self, bot):
+        self.bot = bot
+
+    @commands.command()
+    async def convert(self, ctx, arg: discord.Message, arg2=None):
```
Comment (XuaTheGrate) [inline cogs/General.py]:
```suggestion
    async def convert(self, ctx, message: discord.Message, unit=None):
```

File: cogs/General.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        """
+        # You can tell Sleigh worked on this command...
+        # TODO: All messages here should be constructed and sent as an embed. I'm too lazy to do it today. Tomorrow.
+        #  Soon. Eventually.
+
+        # TODO: Make this a bot var
+        temp_default = True
+
+        # I have no fucking clue how RegEx works
+        result = re.search(r"(\d+)(?:[\s*°]?(?:[degrs]?)*)\s?([fc])?", arg.content, flags=re.I)
```
Comment (XuaTheGrate) [inline cogs/General.py]:
```suggestion
        result = re.search(r"(\d+)(?:[\s*°]?(?:[degrs]?)*)\s?([fc])?", message.content, flags=re.I)
```

File: cogs/General.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+        # No results
+        if result is None:
+            ctx.send("Error: No match found.")
```
Comment (XuaTheGrate) [inline cogs/General.py]:
```suggestion
            await ctx.send("Error: No match found.")
```

File: cogs/General.py [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+        # An else *should* be enough here? Because if it doesn't match then it isn't a number idk
+        else:
+            # Temperature
+            # Formula: T(°C) = (T(°F) - 32) / 1.8
+            temp = int(result[2])
+
+            # C to F
+            if result[2].lower() == "c" or not temp_default:
+                ctx.send(f"{temp}°C is roughly {round((temp * 1.8 + 32), 1)}°F.")
```
Comment (XuaTheGrate) [inline cogs/General.py:42]:
```suggestion
                await ctx.send(f"{temp}°C is roughly {round((temp * 1.8 + 32), 1)}°F.")
```

File: cogs/General.py [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+            # F to C
+            else:
+                ctx.send(f"{temp}°F is roughly {round(((temp - 32) / 1.8), 1)}°C.")
```
Comment (XuaTheGrate) [inline cogs/General.py]:
```suggestion
                await ctx.send(f"{temp}°F is roughly {round(((temp - 32) / 1.8), 1)}°C.")
```

**Review by XuaTheGrate — CHANGES_REQUESTED** [review]
good work!

**Pull request merged**